Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp

```diff
@@ -589,18 +586,96 @@ PlatformImagePtr ImageDecoderCG::createFrameImageAtIndex(size_t index, Subsampli
     return maskedImage ? maskedImage : image;
 }
 
+String ImageDecoderCG::decodeUTI(const SharedBuffer& data) const
+{
+    auto uti = String(CGImageSourceGetType(m_nativeDecoder.get()));
+    if (uti != "public.heif"_s && uti != "public.heic"_s)
+        return uti;
+
+    if (data.size() < sizeof(unsigned)) {
+        ASSERT_NOT_REACHED();
+        return uti;
+    }
+
+    static constexpr auto ftypSignature = FourCC("ftyp");
+    static constexpr auto avifBrand = FourCC("avif");
+    static constexpr auto avisBrand = FourCC("avis");
+    
+    auto boxUnsigned = [&data](unsigned index) -> unsigned {
+        static constexpr bool littleEndian = false;
+        const unsigned* boxBytes = reinterpret_cast<const unsigned*>(data.data());
```

Comment:
I think this is technically UB and probably should be done with bit manipulation instead.